Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoritesView.cpp

```diff
@@ -0,0 +1,179 @@
+
+            connect(m_delegate.data(), &EntryDelegate::RenameEntry, this, &AssetBrowserFavoritesView::AfterRename);
+            connect(this, &QTreeView::customContextMenuRequested, this, &AssetBrowserFavoritesView::OnContextMenu);
+        }
+
+        void AssetBrowserFavoritesView::SetSearchWidget(SearchWidget* searchWidget)
+        {
+            m_favoritesModel.data()->SetSearchWidget(searchWidget);
+        }
+
+        void AssetBrowserFavoritesView::OnContextMenu([[maybe_unused]] const QPoint& point)
+        {
+            auto selectedAssets = selectedIndexes();
+
+            if (selectedAssets.isEmpty())
+            {
+                return;
+            }
+
+            QModelIndex selected = selectedAssets.at(0);
+
+            if (!selected.parent().isValid())
+            {
+                // This is the "favorites" folder.
+                return;
+            }
+
+            AssetBrowserFavoriteItem* favoriteItem = static_cast<AssetBrowserFavoriteItem*>(selected.internalPointer());
+
+            AssetBrowserFavoriteItem::FavoriteType favoriteType = favoriteItem->GetFavoriteType();
+            QMenu menu(this);
+
+            if (favoriteType == AssetBrowserFavoriteItem::FavoriteType::AssetBrowserEntry)
+            {
+                AZStd::vector<const AssetBrowserEntry*> selectedEntries;
+
+                EntryAssetBrowserFavoriteItem* favoriteEntry = static_cast<EntryAssetBrowserFavoriteItem*>(favoriteItem);
+                selectedEntries.push_back(favoriteEntry->GetEntry());
+
+                AssetBrowserInteractionNotificationBus::Broadcast(
+                    &AssetBrowserInteractionNotificationBus::Events::AddContextMenuActions, this, &menu, selectedEntries);
+            }
+            else
+            {
+                if (!m_favoritesModel.data()->GetIsSearchDisabled())
+                {
+                    menu.addAction(
+                        QObject::tr("Apply to Asset Browser"),
+                        [this, favoriteItem]()
+                        {
+                            m_favoritesModel.data()->Select(favoriteItem);
+                        });
+                    menu.addAction(
+                        QObject::tr("Update"),
+                        [this, favoriteItem]()
+                        {
+                            SearchAssetBrowserFavoriteItem* searchFavorite = static_cast<SearchAssetBrowserFavoriteItem*>(favoriteItem);
+                            m_favoritesModel.data()->UpdateSearchItem(searchFavorite);
+                        });
+                }
+                menu.addAction(
+                    QObject::tr("Remove from Favorites"),
+                    [favoriteItem]()
+                    {
+                        AssetBrowserFavoriteRequestBus::Broadcast(&AssetBrowserFavoriteRequestBus::Events::RemoveFromFavorites, favoriteItem);
+                    });
+
+                menu.addAction(
+                    QObject::tr("Rename"),
```

Comment:
nit: "Rename Saved Search" would be more clear here as well

Reply:
Changed